### PR TITLE
added datadoc back from git history

### PR DESCRIPTION
**Pull request by ssb-jnk**

### CODE REVIEW

File: charts/datadoc/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#
+# Library charts provide useful utilities or functions for the chart developer. They're included as
+# a dependency of application charts to inject those utilities and functions into the rendering
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.2.16
```
Comment (Andilun) [inline charts/datadoc/Chart.yaml]: Might conflict with the old release: https://github.com/statisticsnorway/dapla-lab-helm-charts-services/releases/tag/datadoc-0.2.16

File: charts/datadoc/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+| podAnnotations | object | `{}` |  |
+| podLabels."onyxia.app" | string | `"datadoc"` |  |
+| podSecurityContext.fsGroup | int | `100` |  |
+| replicaCount | int | `1` |  |
+| resources | object | `{}` |  |
+| security.allowlist.enabled | bool | `false` |  |
+| security.allowlist.ip | string | `"0.0.0.0/0"` |  |
+| security.networkPolicy.enabled | bool | `false` |  |
+| security.networkPolicy.from | list | `[]` |  |
+| security.oauth2.enabled | bool | `false` |  |
```
Comment (Andilun) [inline charts/datadoc/README.md]: Should be removed.

File: charts/datadoc/templates/oauth2.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
+{{- if .Values.security.oauth2.enabled }}
```
Comment (Andilun) [inline charts/datadoc/templates/oauth2.yaml]: Same, we should remove this and all refs to it.

File: charts/datadoc/templates/statefulset.yaml [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+      labels:
+        {{- include "library-chart.selectorLabels" . | nindent 8 }}
+        {{- with .Values.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    spec:
+      volumes:
+        {{- if .Values.security.oauth2.enabled }}
```
Comment (Andilun) [inline charts/datadoc/templates/statefulset.yaml]: This block should be removed.
{{- if .Values.security.oauth2.enabled }}

File: charts/datadoc/templates/statefulset.yaml [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+              port: {{ .Values.networking.service.port }}
+          startupProbe:
+            httpGet:
+              path: /healthz/startup
+              port: {{ .Values.networking.service.port }}
+            {{- toYaml .Values.startupProbe | nindent 12 }}
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+
+        {{- if .Values.security.oauth2.enabled }}
```
Comment (Andilun) [inline charts/datadoc/templates/statefulset.yaml]: Same with this block.
{{- if .Values.security.oauth2.enabled }}

File: charts/datadoc/values.schema.json [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+            "enabled": {
+              "type": "boolean",
+              "title": "Enable IP protection",
+              "description": "Only the configured set of IPs will be able to reach the service",
+              "default": true,
+              "x-onyxia": {
+                "overwriteDefaultWith": "region.defaultIpProtection"
+              }
+            },
+            "oauth2": {
```
Comment (Andilun) [inline charts/datadoc/values.schema.json]: Should be removed.

File: charts/datadoc/values.yaml [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    enabled: false
+    from: []
+  allowlist:
+    enabled: false
+    ip: "0.0.0.0/0"
+  serviceEntry:
+    enabled: true
+    hosts:
+      - "storage.googleapis.com"
+  oauth2:
```
Comment (Andilun) [inline charts/datadoc/values.yaml:20]: should be removed

File: charts/datadoc/Chart.yaml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+dependencies:
+  - name: library-chart
+    version: 0.2.0
```
Comment (Andilun) [inline charts/datadoc/Chart.yaml]: Not sure if we should do it in this PR, but we should upgrading to the v1.0.1 of the library-chart. So we can use the new auth solution.

File: charts/datadoc/Chart.yaml (hunk 1 above)
Comment (ssb-jnk) [reply charts/datadoc/Chart.yaml]: You're right. I bumped it! :)

**Review by Andilun — APPROVED** [review]

**Pull request merged**